### PR TITLE
First release

### PR DESCRIPTION
**Pull request by diego-antonelli**
Intorducing the video widget for Youtube, Vimeo, Dailymotion and MP4 files
Integration with Pluggable Widgets Tool

### CODE REVIEW

File: src/VideoPlayer.xml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    </icon>
+    <properties>
+        <property key="urlAttribute" type="attribute" required="false">
```
Comment (Andries-Smit) [inline src/VideoPlayer.xml]: Should use Expression, required

File: src/VideoPlayer.xml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            </attributeTypes>
+        </property>
+        <property key="urlStatic" type="string" required="false">
+            <caption>Static URL</caption>
+            <category>General</category>
+            <description>A static url ensures your video will show when there is no value provided from an attribute.
+                When there is a URL provided by an attribute, that will override the static URL.
+            </description>
+        </property>
+        <property key="posterAttribute" type="attribute" required="false">
```
Comment (Andries-Smit) [inline src/VideoPlayer.xml]: Should use expression

File: src/VideoPlayer.xml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            <caption>Poster Static URL</caption>
+            <category>General</category>
+            <description/>
+        </property>
+        <property key="autoStart" type="boolean" defaultValue="false">
+            <caption>Auto start</caption>
+            <category>Behavior</category>
+            <description>Automatically start playing the video when the page loads</description>
+        </property>
+        <property key="showControls" type="boolean" defaultValue="true">
```
Comment (Andries-Smit) [inline src/VideoPlayer.xml]: Can we do this better way with the new group functions (we dont need to release it till WM is fixed)

File: src/components/Dailymotion.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        const attributes = this.handleAttributes();
+        try {
+            if (url.includes("dailymotion.com/embed")) {
+                return `${url}${attributes}`;
+            }
+
+            const urlVimeoSplit = url.split("/");
+            if (urlVimeoSplit.length > 0) {
+                const id = urlVimeoSplit[urlVimeoSplit.length - 1];
+                if (id) return `https://www.dailymotion.com/embed/video/${id}${attributes}`;
```
Comment (Andries-Smit) [inline src/components/Dailymotion.tsx]: No single line if statements, please

**Review by Andries-Smit — COMMENTED** [review]
Should we use expression and property grouping?

File: src/VideoPlayer.xml (hunk 3 above)
Comment (diego-antonelli) [reply src/VideoPlayer.xml]: Done, please review again

File: src/VideoPlayer.xml (hunk 2 above)
Comment (diego-antonelli) [reply src/VideoPlayer.xml]: Done, please review again

File: src/components/Dailymotion.tsx (hunk 4 above)
Comment (diego-antonelli) [reply src/components/Dailymotion.tsx]: Done, please review again

File: src/components/Dailymotion.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    private getUrlAttributes(): string {
+        let attributes = "?sharing-enable=false";
+        attributes += "&autoplay=" + (this.props.autoPlay ? "true" : "false");
+        attributes += "&mute=" + (this.props.muted ? "true" : "false");
+        attributes += "&controls=" + (this.props.showControls ? "true" : "false");
+        return attributes;
+    }
+
+    public static canPlay(url: string): boolean {
+        if (url && validateUrl(url) && url.indexOf("dailymotion.com") > -1) {
```
Comment (Andries-Smit) [inline src/components/Dailymotion.tsx]: one-liner?
``` ts
return url && validateUrl(url) && url.indexOf("dailymotion.com") > -1;
```

Comment (diego-antonelli) [reply src/components/Dailymotion.tsx]: Please review again

File: src/VideoPlayer.xml (hunk 1 above)
Comment (diego-antonelli) [reply src/VideoPlayer.xml]: Done, please review

**Review by Andries-Smit — APPROVED** [review]
Let go!

**Pull request merged**